File: back/src/main/java/com/funstat/store/MdStorage.java
package com.funstat.store;

import com.funstat.domain.Ohlc;
import com.funstat.domain.InstrId;
import firelib.common.interval.Interval;

import java.util.List;

public interface MdStorage {
    List<Ohlc> read(InstrId instrId, String interval);

    void save(String code, String source, String interval, List<firelib.domain.Ohlc> data);

    List<InstrId> getMeta();

    void updateSymbolsMeta();
}

File: back/src/main/java/com/funstat/store/MdStorage.kt
package com.funstat.store

import com.funstat.domain.InstrId
import firelib.common.core.SourceName
import firelib.common.interval.Interval
import firelib.domain.Ohlc

interface MdStorage {

    fun meta(): List<InstrId>
    fun read(instrId: InstrId, interval: Interval): List<Ohlc>

    fun save(code: String, source: SourceName, interval: Interval, data: List<Ohlc>)

    fun updateRequested(code: String)
}


